pkg/gpumanagementfilter/gpu_management_filter.go

```diff
@@ -0,0 +1,129 @@
+package gpumanagementfilter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+const (
+	// PluginName is the name of the plugin used in the plugin registry and configurations.
+	PluginName = "GPUManagementFilter"
+
+	//preFilterStateKey is the key in CycleState to NodeResourcesFit pre-computed data.
+	// Using the name of the plugin helps avoid collisions with other plugins.
+	preFilterStateKey = "PreFilter" + PluginName
+
+	//devicePluginContainerImage is the nvidia provided kubernetes device plugin image.
+	devicePluginContainerImage = "nvcr.io/nvidia/k8s-device-plugin"
+
+	// ResourceNvidiaGPU is nvidia GPU resource
+	ResourceNvidiaGPU = "nvidia.com/gpu"
```

Comment:
I think we should move both of those values so they can be changed via configuration file. Some clusters may also have more than one gpu resource on the node, so it would be great if we could provide a list of gpu resources.

Reply:
+1, Using configuration files is better than hard-coding. Maybe we don't need to limit ourselves to GPU resources, can we extend it to different resources?